tests/init.go

```diff
@@ -0,0 +1,107 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package tests
+
+import (
+	"context"
+	"flag"
+	"google.golang.org/grpc"
+	"math/rand"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	dbaasClient "github.com/percona-platform/dbaas-api/gen/controller"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+//nolint:gochecknoglobals
+var (
+	// Context is canceled on SIGTERM or SIGINT. Tests should cleanup and exit.
+	Context context.Context
+
+	// True if -debug or -trace flag is passed.
+	Debug bool
+
+	// XtraDBClusterAPIClient contains client for dbaas-controller API.
+	XtraDBClusterAPIClient dbaasClient.XtraDBClusterAPIClient
+)
+
+//nolint:gochecknoinits
+func init() {
+	rand.Seed(time.Now().UnixNano())
+
+	debugF := flag.Bool("dbaas.debug", false, "Enable debug output [DBAAS_DEBUG].")
+	traceF := flag.Bool("dbaas.trace", false, "Enable trace output [DBAAS_TRACE].")
+	serverURLF := flag.String("dbaas.server-url", "127.0.0.1:20201", "DBaas Controller URL [DBAAS_SERVER_URL].")
+
+	testing.Init()
+	flag.Parse()
+
+	for envVar, f := range map[string]*flag.Flag{
+		"DBAAS_DEBUG":      flag.Lookup("dbaas.debug"),
+		"DBAAS_TRACE":      flag.Lookup("dbaas.trace"),
+		"DBAAS_SERVER_URL": flag.Lookup("dbaas.server-url"),
+	} {
+		env, ok := os.LookupEnv(envVar)
+		if ok {
+			err := f.Value.Set(env)
+			if err != nil {
+				logrus.Fatalf("Invalid ENV variable %s: %s", envVar, env)
+			}
+		}
+	}
+
+	if *debugF {
+		logrus.SetLevel(logrus.DebugLevel)
```

Comment:
should we reuse logger package from the same repository or let logrus be logger for tests?